plugins/inputs/leofs/leofs.go

```diff
@@ -155,28 +153,19 @@ func (l *LeoFS) Gather(acc telegraf.Accumulator) error {
 		return nil
 	}
 	var wg sync.WaitGroup
-	for i, endpoint := range l.Servers {
-		if !strings.HasPrefix(endpoint, "udp://") {
-			// Preserve backwards compatibility for hostnames without a
-			// scheme, broken in go 1.8. Remove in Telegraf 2.0
-			endpoint = "udp://" + endpoint
-			log.Printf("W! [inputs.mongodb] Using %q as connection URL; please update your configuration to use an URL", endpoint)
-			l.Servers[i] = endpoint
-		}
-		u, err := url.Parse(endpoint)
-		if err != nil {
-			acc.AddError(fmt.Errorf("Unable to parse address %q: %s", endpoint, err))
-			continue
-		}
-		if u.Host == "" {
+	for _, endpoint := range l.Servers {
+		results := strings.Split(endpoint, ":")
+
+		if len(results) > 2 {
 			acc.AddError(fmt.Errorf("Unable to parse address %q", endpoint))
 			continue
 		}
 
-		port := u.Port()
-		if port == "" {
-			port = "4020"
+		port := "4020"
+		if _, err := strconv.Atoi(results[1]); err == nil {
```

Comment:
Since this is an error in the configuration file, I think we should `acc.AddError` and continue if the port cannot be parsed.